### PR TITLE
MM-52352: Limit channel search results

### PR DESCRIPTION
**Pull request by agnivade**
https://mattermost.atlassian.net/browse/MM-52352

```release-note
NONE
```

### CODE REVIEW

**Comment by Willyfrog** [comment]
@harshilsharma63 FYI

File: server/channels/store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -3071,7 +3072,7 @@ func (s SqlChannelStore) Autocomplete(userID, term string, includeDeleted, isGue
 	}
 
 	channels := model.ChannelListWithTeamData{}
-	err = s.GetReplicaX().Select(&channels, sql, args...)
+	err = s.GetSearchReplicaX().Select(&channels, sql, args...)
```
Comment (lieut-data) [inline server/channels/store/sqlstore/channel_store.go:3075]: To confirm, the search replica in a non-ES deployment is "expected" to be hammered by post search, with the theory being that if that gets offloaded (but is still slow), the rest of the application will feel more responsive.

But this particular channel search operation powers the quick switcher, which I imagine gets used at least an order of magnitude more often. Are we confident we should offload this operation to the search replicas as well?

**Review by lieut-data — APPROVED** [review]
Thank you for jumping on this, @agnivade! Server-side changes look solid. One question below about the search replicas.

Also, I imagine the end user experience might degrade /if/ the user expects to start a search and scroll to their matching result. And if the matches exceed 50 channels, it might be harder to select the desired channel. Not a blocker, of course, just thinking out loud.

**Comment by agnivade** [comment]
> Also, I imagine the end user experience might degrade /if/ the user expects to start a search and scroll to their matching result. And if the matches exceed 50 channels, it might be harder to select the desired channel. Not a blocker, of course, just thinking out loud.

Oh yes of course, this needs to be paginated. But that needs to be changed within the product. Right now, all search results have a hard limit, so this change just makes everything a bit consistent.

File: server/channels/store/sqlstore/channel_store.go (hunk 1 above)
Comment (agnivade) [reply server/channels/store/sqlstore/channel_store.go:3075]: Okay that's a good point. I'll revert this change.

**Comment by agnivade** [comment]
No unexpected failures. Merging.

**Pull request merged**